### PR TITLE
tokenize: Don't call str() on dict values

### PR DESCRIPTION
**Pull request by crusaderky**
- XREF https://github.com/dask-contrib/dask-awkward/issues/468

### CODE REVIEW

File: dask/base.py [hunk 1]
```diff
@@ -1062,20 +1062,24 @@ def normalize_type(typ):
...
+    values = _normalize_seq_func(d.values())
+    return "dict", sorted(zip(keys, values), key=lambda kv: str(kv[0]))
 
 
 @normalize_token.register(OrderedDict)
 def normalize_ordered_dict(d):
-    return _normalize_seq_func((type(d), list(d.items())))
+    keys = _normalize_seq_func(d.keys())
+    values = _normalize_seq_func(d.values())
+    return normalize_token(type(d)), list(zip(keys, values))
```
Comment (crusaderky) [inline dask/base.py:1074]: cosmetic; produces a more compact token

**Comment by lgray** [comment]
I think we can drop handling of `key=str` in `set`, at least in my case it will not happen, but it does leave a possibility for future unintended problems like this.

**Comment by crusaderky** [comment]
> I think we can drop handling of `key=str` in `set`, at least in my case it will not happen, but it does leave a possibility for future unintended problems like this.

dropping `key=str` would cause a failure for `tokenize({1, (2, 3)})`.
Ultimately I would much rather work on the assumption that `str()` is cheap for all objects.

**Comment by lgray** [comment]
Sure - I think that's fine and reasonable for set. It may still be useful to stick in the documentation a louder warning/callout/suggestion that `__dask_tokenize__` is very much the preferred option.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @crusaderky

**Pull request merged**